plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -309,25 +242,23 @@ func (t *Tagger) Init() error {
 			t.metadataLookup.instanceId = true
 		case mdKeyImageId:
 			t.metadataLookup.imageId = true
-		case mdKeyInstaneType:
+		case mdKeyInstanceType:
 			t.metadataLookup.instanceType = true
 		default:
-			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s", tag)
+			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s.", tag)
 		}
 	}
 
-	t.Log.Infof("ec2tagger: Check ec2 metadata")
-	if !t.ec2metadata.Available() {
-		msg := "ec2tagger: Unable to retrieve InstanceId. This plugin must only be used on an EC2 instance"
-		t.Log.Errorf(msg)
-		return errors.New(msg)
-	}
+	t.Log.Infof("ec2tagger: EC2 IMDS has started initialization.")
 
-	doc, err := t.ec2metadata.GetInstanceIdentityDocument()
-	if nil != err {
-		msg := fmt.Sprintf("ec2tagger: Unable to retrieve InstanceId : %+v", err.Error())
-		t.Log.Errorf(msg)
-		return errors.New(msg)
+	md := t.ec2MetadataProvider()
+	doc, err := md.GetInstanceIdentityDocument()
```

Comment:
Nit: `md` only gets used once, and that's in the following line. There's really no reason to create the extra variable. This could be consolidated to:
```suggestion
	doc, err := t.ec2MetadataProvider().GetInstanceIdentityDocument()
```